crates/astria-core-utils/src/base64/error.rs

```diff
@@ -0,0 +1,79 @@
+use std::{
+    error,
+    fmt::{
+        self,
+        Display,
+        Formatter,
+    },
+};
+
+/// Errors that can occur while decoding.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Error {
```

Comment:
Let's not make the variants part of the public interface for now? If this just becomes a `pub struct Error(InnerError)` we also don't need to wrap all the `base64` decode errors.

Also call this `DecodeError`?